src/discord/events/message/registerTeam.ts

```diff
@@ -0,0 +1,214 @@
+import Discord from 'discord.js';
+
+import { Config } from '../../../entities/config';
+import logger from '../../../logger';
+import { colors } from '../../constants';
+import { Team } from '../../../entities/team';
+
+class TeamInfo {
+  idArr: string[] = [];
+
+  name = '';
+
+  description = '';
+
+  table = 0;
+
+  user = '';
+
+  constructor(i: string) {
+    this.user = i;
+  }
+
+  addMembers(mem: string[]): void {
+    this.idArr = mem;
+  }
+
+  addUser(): void {
+    this.idArr.push(this.user);
+  }
+
+  addName(n: string): void {
+    this.name = n;
+  }
+
+  addDescription(d: string): void {
+    this.description = d;
+  }
+
+  addTable(t: number): void {
+    this.table = t;
+  }
+}
+
+let registrationInProgress = false;
+let decrement = false;
+let findName: Team;
+let findTable: Team;
+
+const promptArr = [
+  'Who are you hacking with? You have 5 minutes to send a reply. If you are hacking alone, simply write `none`!\n**Be sure to input their Discord ID in a comma-separated list e.g. 111222333444555,666777888999000,...**',
+  "What's your team/app name? You have 2 minutes to answer.",
+  "What's your table number (e.g. 42)? You have 2 minutes to answer.",
+  'What does your project do? How will it make a difference? What technologies are used? You have 5 minutes to answer.',
+];
+const componentArr = ['members', 'name', 'table', 'description'];
+const timeArr = [300000, 120000, 120000, 300000];
+const currentUsers: string[] = [];
```

Comment:
I see what you're doing here with this value, however, what you're doing is making the application `stateful` in the sense that this file has a chunk of data held in memory that is very important to the user experience. To highly the impact of being stateful, follow this example:
1. User starts into the `!registerTeam` flow
1. Their id is added to `currentUsers`
1. They continue through the necessary fields until they get to description
1. The app restarts (either due to an error in the runtime environment itself or because of a deployment)
1. The user submits their description
1. The app has no knowledge of this user being in the flow and their data is lost

We need to rely on the `discordContext` table and use that to store whether the user is in the flow or not and, if they are, what values they've provided us with so far.